[Automation] java run fails for co.fs2:fs2-core_3:3.0-130-1713a29

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0-130-1713a29" ./gradlew clean javaTest -Pcoordinates="co.fs2:fs2-core_3:2.5.6"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25454136977/job/74679302473
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
===============================================================
 co.fs2:fs2-core_3:2.5.6:3.0-130-1713a29 stage java run
========================================================================================
Starting stage 'java run' for 3.0-130-1713a29 at 2026-05-06T18:55:52Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0-130-1713a29" ./gradlew clean javaTest -Pcoordinates="co.fs2:fs2-core_3:2.5.6"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 718ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava NO-SOURCE

> Task :compileTestScala
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:9:20: value ContextShift is not a member of cats.effect
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:11:20: value Timer is not a member of cats.effect
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:12:20: value concurrent is not a member of cats.effect
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:17:23: value Queue is not a member of fs2.concurrent
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:41:38: Not found: type ContextShift
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:42:31: Not found: type Timer - did you mean Timeout?
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:41:60: value contextShift is not a member of object cats.effect.IO
exception caught when loading module class ModuleUtils$: java.lang.AssertionError: org.junit.platform.commons.util.ModuleUtils$ModuleReferenceScanner
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:42:46: value timer is not a member of object cats.effect.IO
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:48:20: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type Numeric[Int] of a context parameter of [33mmethod[0m [35mrange[0m in [33mobject[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:65:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:74:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:94:94: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Int] of a context parameter of [33mmethod[0m [35marray[0m in [33mobject[0m [35mChunk[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:117:87: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.RaiseThrowable[cats.effect.IO] of a context parameter of [33mmethod[0m [35mraiseError[0m in [33mobject[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:118:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:133:68: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Int] of a context parameter of [33mmethod[0m [35marray[0m in [33mobject[0m [35mChunk[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:143:34: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Int] of parameter evidence$12 of [33mmethod[0m [35mfill[0m in [33mobject[0m [35mArray[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:147:62: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Byte] of parameter evidence$5 of [33mmethod[0m [35mapply[0m in [33mobject[0m [35mArray[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:157:42: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Byte] of parameter evidence$5 of [33mmethod[0m [35mapply[0m in [33mobject[0m [35mArray[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:163:84: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Byte] of parameter evidence$5 of [33mmethod[0m [35mapply[0m in [33mobject[0m [35mArray[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:169:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:177:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:183:34: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.RaiseThrowable[F] of a context parameter of [33mmethod[0m [35mdecode[0m in [33mobject[0m [35mbase64[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:190:55: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[B] of parameter evidence$2 of [33mmethod[0m [35mtoArray[0m in [33mtrait[0m [35mIterableOnceOps[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:200:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:206:28: value deflate is not a member of fs2.compression
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:215:21: value gzip is not a member of fs2.compression
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:233:53: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[B] of parameter evidence$2 of [33mmethod[0m [35mtoArray[0m in [33mtrait[0m [35mIterableOnceOps[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:234:57: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[B] of parameter evidence$2 of [33mmethod[0m [35mtoArray[0m in [33mtrait[0m [35mIterableOnceOps[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:235:53: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[B] of parameter evidence$2 of [33mmethod[0m [35mtoArray[0m in [33mtrait[0m [35mIterableOnceOps[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:278:16: Not found: Queue
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:300:16: Not found: Queue
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:320:63: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type cats.effect.kernel.Concurrent[cats.effect.IO] of parameter F of [33mmethod[0m [35mconstant[0m in [33mobject[0m [35mSignal[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:321:50: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type cats.effect.unsafe.IORuntime of parameter runtime of [33mmethod[0m [35munsafeRunSync[0m in [33mclass[0m [35mIOPlatform[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:322:65: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:331:27: value bytes is not a member of object fs2.Chunk
35 errors found

> Task :compileTestScala FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestScala'.
> Compilation failed

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 8s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
12 actionable tasks: 2 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 10s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 3.0-130-1713a29 at 2026-05-06T18:56:02Z with exit code 1 after 10s
❌ ERROR: 'java run' for 3.0-130-1713a29 failed with exit code 1.
FAILED[java run][3.0-130-1713a29][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0-130-1713a29" ./gradlew clean javaTest -Pcoordinates="co.fs2:fs2-core_3:2.5.6"]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0-130-1713a29" ./gradlew clean javaTest -Pcoordinates="co.fs2:fs2-core_3:2.5.6"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25454136977/job/74679302585
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
==============================================================
 co.fs2:fs2-core_3:2.5.6:3.0-130-1713a29 stage java run
========================================================================================
Starting stage 'java run' for 3.0-130-1713a29 at 2026-05-06T18:56:09Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0-130-1713a29" ./gradlew clean javaTest -Pcoordinates="co.fs2:fs2-core_3:2.5.6"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 751ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :compileJava NO-SOURCE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava NO-SOURCE

> Task :compileTestScala
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:9:20: value ContextShift is not a member of cats.effect
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:11:20: value Timer is not a member of cats.effect
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:12:20: value concurrent is not a member of cats.effect
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:17:23: value Queue is not a member of fs2.concurrent
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:41:38: Not found: type ContextShift
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:42:31: Not found: type Timer - did you mean Timeout?
exception caught when loading module class ModuleUtils$: java.lang.AssertionError: org.junit.platform.commons.util.ModuleUtils$ModuleReferenceScanner
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:41:60: value contextShift is not a member of object cats.effect.IO
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:42:46: value timer is not a member of object cats.effect.IO
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:48:20: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type Numeric[Int] of a context parameter of [33mmethod[0m [35mrange[0m in [33mobject[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:65:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:74:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:94:94: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Int] of a context parameter of [33mmethod[0m [35marray[0m in [33mobject[0m [35mChunk[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:117:87: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.RaiseThrowable[cats.effect.IO] of a context parameter of [33mmethod[0m [35mraiseError[0m in [33mobject[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:118:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:133:68: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Int] of a context parameter of [33mmethod[0m [35marray[0m in [33mobject[0m [35mChunk[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:143:34: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Int] of parameter evidence$12 of [33mmethod[0m [35mfill[0m in [33mobject[0m [35mArray[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:147:62: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Byte] of parameter evidence$5 of [33mmethod[0m [35mapply[0m in [33mobject[0m [35mArray[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:157:42: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Byte] of parameter evidence$5 of [33mmethod[0m [35mapply[0m in [33mobject[0m [35mArray[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:163:84: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Byte] of parameter evidence$5 of [33mmethod[0m [35mapply[0m in [33mobject[0m [35mArray[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:169:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:177:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:183:34: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.RaiseThrowable[F] of a context parameter of [33mmethod[0m [35mdecode[0m in [33mobject[0m [35mbase64[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:190:55: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[B] of parameter evidence$2 of [33mmethod[0m [35mtoArray[0m in [33mtrait[0m [35mIterableOnceOps[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:200:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:206:28: value deflate is not a member of fs2.compression
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:215:21: value gzip is not a member of fs2.compression
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:233:53: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[B] of parameter evidence$2 of [33mmethod[0m [35mtoArray[0m in [33mtrait[0m [35mIterableOnceOps[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:234:57: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[B] of parameter evidence$2 of [33mmethod[0m [35mtoArray[0m in [33mtrait[0m [35mIterableOnceOps[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:235:53: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[B] of parameter evidence$2 of [33mmethod[0m [35mtoArray[0m in [33mtrait[0m [35mIterableOnceOps[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:278:16: Not found: Queue
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:300:16: Not found: Queue
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:320:63: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type cats.effect.kernel.Concurrent[cats.effect.IO] of parameter F of [33mmethod[0m [35mconstant[0m in [33mobject[0m [35mSignal[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:321:50: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type cats.effect.unsafe.IORuntime of parameter runtime of [33mmethod[0m [35munsafeRunSync[0m in [33mclass[0m [35mIOPlatform[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:322:65: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:331:27: value bytes is not a member of object fs2.Chunk
35 errors found

> Task :compileTestScala FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/build/reports/problems/problems-report.html


Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
12 actionable tasks: 2 executed, 10 up-to-date
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestScala'.
> Compilation failed

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 8s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 10s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 3.0-130-1713a29 at 2026-05-06T18:56:19Z with exit code 1 after 10s
❌ ERROR: 'java run' for 3.0-130-1713a29 failed with exit code 1.
FAILED[java run][3.0-130-1713a29][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.0-130-1713a29" ./gradlew clean javaTest -Pcoordinates="co.fs2:fs2-core_3:2.5.6"]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.0-130-1713a29" ./gradlew clean javaTest -Pcoordinates="co.fs2:fs2-core_3:2.5.6"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25454136977/job/74679302543
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
===============================================================
 co.fs2:fs2-core_3:2.5.6:3.0-130-1713a29 stage java run
========================================================================================
Starting stage 'java run' for 3.0-130-1713a29 at 2026-05-06T18:56:48Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.0-130-1713a29" ./gradlew clean javaTest -Pcoordinates="co.fs2:fs2-core_3:2.5.6"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 724ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :compileScala NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava NO-SOURCE

> Task :compileTestScala
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:9:20: value ContextShift is not a member of cats.effect
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:11:20: value Timer is not a member of cats.effect
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:12:20: value concurrent is not a member of cats.effect
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:17:23: value Queue is not a member of fs2.concurrent
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:41:38: Not found: type ContextShift
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:42:31: Not found: type Timer - did you mean Timeout?
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:41:60: value contextShift is not a member of object cats.effect.IO
exception caught when loading module class ModuleUtils$: java.lang.AssertionError: org.junit.platform.commons.util.ModuleUtils$ModuleReferenceScanner
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:42:46: value timer is not a member of object cats.effect.IO
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:48:20: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type Numeric[Int] of a context parameter of [33mmethod[0m [35mrange[0m in [33mobject[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:65:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:74:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:94:94: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Int] of a context parameter of [33mmethod[0m [35marray[0m in [33mobject[0m [35mChunk[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:117:87: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.RaiseThrowable[cats.effect.IO] of a context parameter of [33mmethod[0m [35mraiseError[0m in [33mobject[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:118:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:133:68: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Int] of a context parameter of [33mmethod[0m [35marray[0m in [33mobject[0m [35mChunk[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:143:34: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Int] of parameter evidence$12 of [33mmethod[0m [35mfill[0m in [33mobject[0m [35mArray[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:147:62: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Byte] of parameter evidence$5 of [33mmethod[0m [35mapply[0m in [33mobject[0m [35mArray[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:157:42: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Byte] of parameter evidence$5 of [33mmethod[0m [35mapply[0m in [33mobject[0m [35mArray[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:163:84: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[Byte] of parameter evidence$5 of [33mmethod[0m [35mapply[0m in [33mobject[0m [35mArray[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:169:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:177:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:183:34: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.RaiseThrowable[F] of a context parameter of [33mmethod[0m [35mdecode[0m in [33mobject[0m [35mbase64[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:190:55: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[B] of parameter evidence$2 of [33mmethod[0m [35mtoArray[0m in [33mtrait[0m [35mIterableOnceOps[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:200:15: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:206:28: value deflate is not a member of fs2.compression
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:215:21: value gzip is not a member of fs2.compression
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:233:53: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[B] of parameter evidence$2 of [33mmethod[0m [35mtoArray[0m in [33mtrait[0m [35mIterableOnceOps[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:234:57: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[B] of parameter evidence$2 of [33mmethod[0m [35mtoArray[0m in [33mtrait[0m [35mIterableOnceOps[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:235:53: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type scala.reflect.ClassTag[B] of parameter evidence$2 of [33mmethod[0m [35mtoArray[0m in [33mtrait[0m [35mIterableOnceOps[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:278:16: Not found: Queue
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:300:16: Not found: Queue
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:320:63: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type cats.effect.kernel.Concurrent[cats.effect.IO] of parameter F of [33mmethod[0m [35mconstant[0m in [33mobject[0m [35mSignal[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:321:50: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type cats.effect.unsafe.IORuntime of parameter runtime of [33mmethod[0m [35munsafeRunSync[0m in [33mclass[0m [35mIOPlatform[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:322:65: Ambiguous given instances: both [33mvalue[0m [35mcontextShift[0m in [33mclass[0m [35mFs2_core_3Test[0m and [33mvalue[0m [35mtimer[0m in [33mclass[0m [35mFs2_core_3Test[0m match type fs2.Compiler[F2, G] of parameter compiler of [33mmethod[0m [35mcompile[0m in [33mclass[0m [35mStream[0m
[Error] /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala:331:27: value bytes is not a member of object fs2.Chunk
35 errors found

> Task :compileTestScala FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/co.fs2/fs2-core_3/2.5.6/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
12 actionable tasks: 2 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestScala'.
> Compilation failed

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 8s

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 10s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 3.0-130-1713a29 at 2026-05-06T18:56:58Z with exit code 1 after 10s
❌ ERROR: 'java run' for 3.0-130-1713a29 failed with exit code 1.
FAILED[java run][3.0-130-1713a29][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.0-130-1713a29" ./gradlew clean javaTest -Pcoordinates="co.fs2:fs2-core_3:2.5.6"]
```

--- a/metadata/co.fs2/fs2-core_3/3.0-130-1713a29/reachability-metadata.json
+++ b/metadata/co.fs2/fs2-core_3/3.0-130-1713a29/reachability-metadata.json
@@ -1,0 +1,48 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "fs2.internal.ScopedResource$"
+      },
+      "type": "fs2.internal.ScopedResource$$anon$1",
+      "fields": [
+        {
+          "name": "0bitmap$1"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "scodec.bits.ByteVector$"
+      },
+      "type": "scodec.bits.ByteVector",
+      "fields": [
+        {
+          "name": "0bitmap$1"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "fs2.Chunk"
+      },
+      "type": "fs2.Chunk$",
+      "fields": [
+        {
+          "name": "0bitmap$1"
+        }
+      ]
+    },
+    {
+      "condition": {
+        "typeReached": "fs2.compression.Compression$$anon$1"
+      },
+      "type": "fs2.compression.Compression$$anon$1",
+      "fields": [
+        {
+          "name": "0bitmap$1"
+        }
+      ]
+    }
+  ]
+}

--- a/metadata/co.fs2/fs2-core_3/index.json
+++ b/metadata/co.fs2/fs2-core_3/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "3.0-130-1713a29",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/co/fs2/fs2-core_3/$version$/fs2-core_3-$version$-sources.jar",
+    "repository-url" : "https://github.com/typelevel/fs2",
+    "test-code-url" : "https://github.com/typelevel/fs2/tree/1713a29/core/shared/src/test",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/co/fs2/fs2-core_3/$version$/fs2-core_3-$version$-javadoc.jar",
+    "description" : "FS2 Core is the core module of Functional Streams for Scala, providing purely functional, effectful, and polymorphic stream processing abstractions. It defines composable streams and pulls with resource-safe execution, transformation, concurrency, and integration with Cats Effect typeclasses.",
     "language" : {
       "name" : "scala",
       "version" : "3"

--- a/metadata/co.fs2/fs2-core_3/index.json
+++ b/metadata/co.fs2/fs2-core_3/index.json
@@ -1,6 +1,20 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "3.0-130-1713a29",
+    "language" : {
+      "name" : "scala",
+      "version" : "3"
+    },
+    "tested-versions" : [
+      "3.0-130-1713a29"
+    ],
+    "allowed-packages" : [
+      "fs2",
+      "scodec.bits"
+    ]
+  },
+  {
     "metadata-version" : "2.5.6",
     "source-code-url" : "https://repo.maven.apache.org/maven2/co/fs2/fs2-core_3/$version$/fs2-core_3-$version$-sources.jar",
     "repository-url" : "https://github.com/typelevel/fs2",

--- a/stats/co.fs2/fs2-core_3/3.0-130-1713a29/execution-metrics.json
+++ b/stats/co.fs2/fs2-core_3/3.0-130-1713a29/execution-metrics.json
@@ -1,0 +1,87 @@
+{
+  "fix_java_run_fail:2026-05-07": {
+    "timestamp": "2026-05-07T08:22:00.594056Z",
+    "library": "co.fs2:fs2-core_3:3.0-130-1713a29",
+    "starting_commit": "a2d500d73ac613177d4980d175c833e456cfafc4",
+    "ending_commit": "15b0499e3b6df77d98d215cbbadb50f7dc42a2c9",
+    "previous_library": "co.fs2:fs2-core_3:2.5.6",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "3.0-130-1713a29",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 11079,
+          "missed": 25516,
+          "ratio": 0.302746,
+          "total": 36595
+        },
+        "line": {
+          "covered": 1565,
+          "missed": 1726,
+          "ratio": 0.475539,
+          "total": 3291
+        },
+        "method": {
+          "covered": 954,
+          "missed": 2674,
+          "ratio": 0.262955,
+          "total": 3628
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "2.5.6",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 11482,
+          "missed": 35982,
+          "ratio": 0.24191,
+          "total": 47464
+        },
+        "line": {
+          "covered": 1636,
+          "missed": 2361,
+          "ratio": 0.409307,
+          "total": 3997
+        },
+        "method": {
+          "covered": 1028,
+          "missed": 3886,
+          "ratio": 0.209198,
+          "total": 4914
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 47695,
+      "cached_input_tokens_used": 909824,
+      "output_tokens_used": 5874,
+      "iterations": 1,
+      "cost_usd": 0.6311,
+      "tested_library_loc": 1565,
+      "metadata_entries": 8,
+      "previous_library_metadata_entries": 4,
+      "code_coverage_percent": 47.55,
+      "previous_library_coverage_percent": 40.93
+    },
+    "artifacts": {
+      "test_file": "tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala",
+      "metadata_file": "metadata/co.fs2/fs2-core_3/3.0-130-1713a29/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/co.fs2/fs2-core_3/3.0-130-1713a29/stats.json
+++ b/stats/co.fs2/fs2-core_3/3.0-130-1713a29/stats.json
@@ -1,0 +1,31 @@
+{
+  "versions" : [ {
+    "version" : "3.0-130-1713a29",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 11079,
+        "missed" : 25516,
+        "ratio" : 0.302746,
+        "total" : 36595
+      },
+      "line" : {
+        "covered" : 1565,
+        "missed" : 1726,
+        "ratio" : 0.475539,
+        "total" : 3291
+      },
+      "method" : {
+        "covered" : 954,
+        "missed" : 2674,
+        "ratio" : 0.262955,
+        "total" : 3628
+      }
+    }
+  } ]
+}

--- a/tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/.gitignore
+++ b/tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/build.gradle
+++ b/tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/build.gradle
@@ -1,0 +1,36 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+    id "scala"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "co.fs2:fs2-core_3:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+    testImplementation 'org.scala-lang:scala3-library_3:3.3.6'
+    testImplementation 'org.scala-lang:scala3-compiler_3:3.3.6'
+}
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(21)
+    }
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/gradle.properties
+++ b/tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = co.fs2:fs2-core_3:3.0-130-1713a29
+library.version = 3.0-130-1713a29
+metadata.dir = co.fs2/fs2-core_3/3.0-130-1713a29/

--- a/tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/settings.gradle
+++ b/tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'co.fs2.fs2-core_3_tests'

--- a/tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala
+++ b/tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala
@@ -1,0 +1,335 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package co_fs2.fs2_core_3
+
+import cats.effect.ContextShift
+import cats.effect.IO
+import cats.effect.Timer
+import cats.effect.concurrent.Deferred
+import fs2.Chunk
+import fs2.Pull
+import fs2.Stream
+import fs2.compression
+import fs2.concurrent.Queue
+import fs2.concurrent.Signal
+import fs2.hash
+import fs2.text
+import org.junit.jupiter.api.Assertions.assertArrayEquals
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertSame
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Assertions.fail
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.Timeout
+
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets
+import java.security.MessageDigest
+import java.time.Instant
+import java.util.Base64
+import java.util.concurrent.CopyOnWriteArrayList
+import java.util.concurrent.TimeUnit
+import scala.concurrent.ExecutionContext
+import scala.jdk.CollectionConverters.*
+
+class Fs2_core_3Test {
+  private implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+  private implicit val timer: Timer[IO] = IO.timer(ExecutionContext.global)
+
+  @Test
+  @Timeout(value = 60, unit = TimeUnit.SECONDS)
+  def streamTransformationsCompileToExpectedCollections(): Unit = {
+    val transformed: Vector[String] = Stream
+      .range(0, 20)
+      .filter(_ % 3 != 0)
+      .map(_ + 1)
+      .drop(2)
+      .take(5)
+      .zipWithIndex
+      .map { case (value, index) => s"$index:$value" }
+      .intersperse("|")
+      .covary[IO]
+      .compile
+      .toVector
+      .unsafeRunSync()
+
+    val sum: Int = Stream
+      .emits(List(1, 2, 3, 4, 5))
+      .scan(0)(_ + _)
+      .covary[IO]
+      .compile
+      .lastOrError
+      .unsafeRunSync()
+
+    val chunked: Vector[List[Int]] = Stream
+      .emits(List(1, 2, 3, 4, 5))
+      .chunkN(2)
+      .map(_.toList)
+      .covary[IO]
+      .compile
+      .toVector
+      .unsafeRunSync()
+
+    assertEquals(Vector("0:5", "|", "1:6", "|", "2:8", "|", "3:9", "|", "4:11"), transformed)
+    assertEquals(15, sum)
+    assertEquals(Vector(List(1, 2), List(3, 4), List(5)), chunked)
+  }
+
+  @Test
+  @Timeout(value = 60, unit = TimeUnit.SECONDS)
+  def pullUnconsumesAndReemitsBoundedChunks(): Unit = {
+    def grouped(source: Stream[IO, Int]): Pull[IO, String, Unit] =
+      source.pull.unconsLimit(3).flatMap {
+        case Some((chunk, tail)) =>
+          Pull.output1(chunk.toList.mkString("[", ",", "]")) >> grouped(tail)
+        case None =>
+          Pull.done
+      }
+
+    val groups: Vector[String] = grouped(Stream.chunk(Chunk.array(Array(1, 2, 3, 4, 5, 6, 7))).covary[IO])
+      .stream
+      .compile
+      .toVector
+      .unsafeRunSync()
+
+    assertEquals(Vector("[1,2,3]", "[4,5,6]", "[7]"), groups)
+  }
+
+  @Test
+  @Timeout(value = 60, unit = TimeUnit.SECONDS)
+  def bracketFinalizerRunsWhenStreamFails(): Unit = {
+    val events = new CopyOnWriteArrayList[String]()
+    val boom = new IllegalStateException("boom")
+
+    val result: Either[Throwable, Vector[Int]] = Stream
+      .bracket(IO {
+        events.add("acquire")
+        "token"
+      })(resource => IO {
+        events.add(s"release:$resource")
+        ()
+      })
+      .flatMap(resource => Stream.emit(resource.length) ++ Stream.raiseError[IO](boom))
+      .compile
+      .toVector
+      .attempt
+      .unsafeRunSync()
+
+    result match {
+      case Left(error) => assertSame(boom, error)
+      case Right(value) => fail(s"Expected stream failure, got $value")
+    }
+    assertEquals(List("acquire", "release:token"), events.asScala.toList)
+  }
+
+  @Test
+  @Timeout(value = 60, unit = TimeUnit.SECONDS)
+  def chunksSupportPrimitiveStorageSlicingAndStatefulMapping(): Unit = {
+    val chunk: Chunk[Int] = Chunk.array(Array(1, 2, 3, 4, 5), 1, 3)
+    val positiveTens: List[Int] = chunk
+      .flatMap(value => Chunk(value, -value))
+      .filter(_ > 0)
+      .map(_ * 10)
+      .toList
+    val (finalState, accumulated): (Int, Chunk[String]) = chunk.mapAccumulate(0) { (sum, value) =>
+      val nextSum = sum + value
+      (nextSum, s"$value->$nextSum")
+    }
+    val copied = Array.fill(5)(0)
+
+    chunk.copyToArray(copied, 1)
+    val bytes: Array[Byte] = Chunk
+      .byteBuffer(ByteBuffer.wrap(Array[Byte](10, 20, 30, 40)))
+      .drop(1)
+      .take(2)
+      .toBytes
+      .toArray
+
+    assertEquals(List(20, 30, 40), positiveTens)
+    assertEquals(9, finalState)
+    assertEquals(List("2->2", "3->5", "4->9"), accumulated.toList)
+    assertArrayEquals(Array(0, 2, 3, 4, 0), copied)
+    assertArrayEquals(Array[Byte](20, 30), bytes)
+  }
+
+  @Test
+  @Timeout(value = 60, unit = TimeUnit.SECONDS)
+  def textPipesHandleUtf8LineSplittingAndBase64(): Unit = {
+    val textBytes: Array[Byte] = Array[Byte](0xef.toByte, 0xbb.toByte, 0xbf.toByte) ++
+      "hé\nworld\r\nlast".getBytes(StandardCharsets.UTF_8)
+
+    val decodedLines: Vector[String] = byteStream(textBytes, 2)
+      .through(text.utf8Decode)
+      .through(text.lines)
+      .compile
+      .toVector
+      .unsafeRunSync()
+
+    val plainText: String = "native-image friendly fs2"
+    val plainBytes: Array[Byte] = plainText.getBytes(StandardCharsets.UTF_8)
+    val encoded: String = byteStream(plainBytes, 5)
+      .through(text.base64.encode)
+      .compile
+      .string
+      .unsafeRunSync()
+    val decodedBytes: Vector[Byte] = Stream
+      .emits(Vector(encoded.take(4), " \n", encoded.drop(4)))
+      .covary[IO]
+      .through(text.base64.decode)
+      .compile
+      .toVector
+      .unsafeRunSync()
+
+    assertEquals(Vector("hé", "world", "last"), decodedLines)
+    assertEquals(Base64.getEncoder.encodeToString(plainBytes), encoded)
+    assertArrayEquals(plainBytes, decodedBytes.toArray)
+  }
+
+  @Test
+  @Timeout(value = 60, unit = TimeUnit.SECONDS)
+  def hashAndCompressionPipesRoundTripChunkedBytes(): Unit = {
+    val payload: Array[Byte] = ("FS2 compression and hashing " * 16).getBytes(StandardCharsets.UTF_8)
+
+    val sha256: Vector[Byte] = byteStream(payload, 11)
+      .through(hash.sha256)
+      .compile
+      .toVector
+      .unsafeRunSync()
+    val expectedDigest: Array[Byte] = MessageDigest.getInstance("SHA-256").digest(payload)
+
+    val deflatedRoundTrip: Vector[Byte] = byteStream(payload, 7)
+      .through(compression.deflate[IO](compression.DeflateParams.BEST_SPEED))
+      .through(compression.inflate[IO]())
+      .compile
+      .toVector
+      .unsafeRunSync()
+
+    val modificationTime: Instant = Instant.parse("2020-01-02T03:04:05Z")
+    val gzipRoundTrip: Vector[Byte] = byteStream(payload, 9)
+      .through(
+        compression.gzip[IO](
+          fileName = Some("payload.txt"),
+          modificationTime = Some(modificationTime),
+          comment = Some("metadata")
+        )
+      )
+      .through(compression.gunzip[IO]())
+      .flatMap { result =>
+        Stream.eval(IO {
+          assertEquals(Some("payload.txt"), result.fileName)
+          assertEquals(Some(modificationTime), result.modificationTime)
+          assertEquals(Some("metadata"), result.comment)
+        }).drain ++ result.content
+      }
+      .compile
+      .toVector
+      .unsafeRunSync()
+
+    assertArrayEquals(expectedDigest, sha256.toArray)
+    assertArrayEquals(payload, deflatedRoundTrip.toArray)
+    assertArrayEquals(payload, gzipRoundTrip.toArray)
+  }
+
+  @Test
+  @Timeout(value = 60, unit = TimeUnit.SECONDS)
+  def mapAsyncRunsEffectsConcurrentlyWhilePreservingInputOrder(): Unit = {
+    val ordered: Vector[Int] = (for {
+      firstReady <- Deferred[IO, Unit]
+      secondReady <- Deferred[IO, Unit]
+      thirdReady <- Deferred[IO, Unit]
+      firstGate <- Deferred[IO, Unit]
+      secondGate <- Deferred[IO, Unit]
+      thirdGate <- Deferred[IO, Unit]
+      readyByValue: Map[Int, Deferred[IO, Unit]] = Map(1 -> firstReady, 2 -> secondReady, 3 -> thirdReady)
+      gateByValue: Map[Int, Deferred[IO, Unit]] = Map(1 -> firstGate, 2 -> secondGate, 3 -> thirdGate)
+      fiber <- Stream
+        .emits(List(1, 2, 3))
+        .covary[IO]
+        .mapAsync(3) { value =>
+          for {
+            _ <- readyByValue(value).complete(())
+            _ <- gateByValue(value).get
+          } yield value
+        }
+        .compile
+        .toVector
+        .start
+      _ <- firstReady.get
+      _ <- secondReady.get
+      _ <- thirdReady.get
+      _ <- thirdGate.complete(())
+      _ <- secondGate.complete(())
+      _ <- firstGate.complete(())
+      values <- fiber.join
+    } yield values).unsafeRunSync()
+
+    assertEquals(Vector(1, 2, 3), ordered)
+  }
+
+  @Test
+  @Timeout(value = 60, unit = TimeUnit.SECONDS)
+  def queuesExposeBoundedOffersChunksAndTermination(): Unit = {
+    val queueResult: (Boolean, Boolean, Boolean, List[Int], Option[Int], Int, Int) = (for {
+      queue <- Queue.bounded[IO, Int](2)
+      firstOffer <- queue.offer1(1)
+      secondOffer <- queue.offer1(2)
+      thirdOffer <- queue.offer1(3)
+      chunk <- queue.dequeueChunk1(5)
+      empty <- queue.tryDequeue1
+      _ <- queue.enqueue1(4)
+      dequeued <- queue.dequeue1
+      mapped = queue.imap[String](_.toString)(_.toInt)
+      _ <- mapped.enqueue1("5")
+      mappedDequeued <- queue.dequeue1
+    } yield (
+      firstOffer,
+      secondOffer,
+      thirdOffer,
+      chunk.toList,
+      empty,
+      dequeued,
+      mappedDequeued
+    )).unsafeRunSync()
+
+    val terminated: Vector[String] = (for {
+      queue <- Queue.noneTerminated[IO, String]
+      _ <- queue.enqueue1(Some("alpha"))
+      _ <- queue.enqueue1(Some("beta"))
+      _ <- queue.enqueue1(None)
+      values <- queue.dequeue.compile.toVector
+    } yield values).unsafeRunSync()
+
+    assertTrue(queueResult._1)
+    assertTrue(queueResult._2)
+    assertFalse(queueResult._3)
+    assertEquals(List(1, 2), queueResult._4)
+    assertEquals(None, queueResult._5)
+    assertEquals(4, queueResult._6)
+    assertEquals(5, queueResult._7)
+    assertEquals(Vector("alpha", "beta"), terminated)
+  }
+
+  @Test
+  @Timeout(value = 60, unit = TimeUnit.SECONDS)
+  def signalConstantCanBeMappedAndSampledContinuously(): Unit = {
+    val signal: Signal[IO, Int] = Signal.constant[IO, Int](21).map(_ * 2)
+    val current: Int = signal.get.unsafeRunSync()
+    val samples: Vector[Int] = signal.continuous.take(3).compile.toVector.unsafeRunSync()
+
+    assertEquals(42, current)
+    assertEquals(Vector(42, 42, 42), samples)
+  }
+
+  private def byteStream(bytes: Array[Byte], chunkSize: Int): Stream[IO, Byte] = {
+    val chunks: Vector[Chunk[Byte]] = bytes
+      .grouped(chunkSize)
+      .map(chunk => Chunk.bytes(chunk))
+      .toVector
+    Stream.emits(chunks).covary[IO].flatMap(chunk => Stream.chunk(chunk).covary[IO])
+  }
+}

--- a/tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala
+++ b/tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/src/test/scala/co_fs2/fs2_core_3/Fs2_core_3Test.scala
@@ -6,15 +6,17 @@
  */
 package co_fs2.fs2_core_3
 
-import cats.effect.ContextShift
+import cats.effect.Deferred
 import cats.effect.IO
-import cats.effect.Timer
-import cats.effect.concurrent.Deferred
+import cats.effect.Outcome
+import cats.effect.std.Queue
+import cats.effect.unsafe.implicits.global
 import fs2.Chunk
 import fs2.Pull
 import fs2.Stream
-import fs2.compression
-import fs2.concurrent.Queue
+import fs2.compression.Compression
+import fs2.compression.DeflateParams
+import fs2.compression.InflateParams
 import fs2.concurrent.Signal
 import fs2.hash
 import fs2.text
@@ -34,13 +36,9 @@ import java.time.Instant
 import java.util.Base64
 import java.util.concurrent.CopyOnWriteArrayList
 import java.util.concurrent.TimeUnit
-import scala.concurrent.ExecutionContext
 import scala.jdk.CollectionConverters.*
 
 class Fs2_core_3Test {
-  private implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
-  private implicit val timer: Timer[IO] = IO.timer(ExecutionContext.global)
-
   @Test
   @Timeout(value = 60, unit = TimeUnit.SECONDS)
   def streamTransformationsCompileToExpectedCollections(): Unit = {
@@ -147,7 +145,6 @@ class Fs2_core_3Test {
       .byteBuffer(ByteBuffer.wrap(Array[Byte](10, 20, 30, 40)))
       .drop(1)
       .take(2)
-      .toBytes
       .toArray
 
     assertEquals(List(20, 30, 40), positiveTens)
@@ -203,8 +200,8 @@ class Fs2_core_3Test {
     val expectedDigest: Array[Byte] = MessageDigest.getInstance("SHA-256").digest(payload)
 
     val deflatedRoundTrip: Vector[Byte] = byteStream(payload, 7)
-      .through(compression.deflate[IO](compression.DeflateParams.BEST_SPEED))
-      .through(compression.inflate[IO]())
+      .through(Compression[IO].deflate(DeflateParams.BEST_SPEED))
+      .through(Compression[IO].inflate(InflateParams.DEFAULT))
       .compile
       .toVector
       .unsafeRunSync()
@@ -212,13 +209,13 @@ class Fs2_core_3Test {
     val modificationTime: Instant = Instant.parse("2020-01-02T03:04:05Z")
     val gzipRoundTrip: Vector[Byte] = byteStream(payload, 9)
       .through(
-        compression.gzip[IO](
+        Compression[IO].gzip(
           fileName = Some("payload.txt"),
           modificationTime = Some(modificationTime),
           comment = Some("metadata")
         )
       )
-      .through(compression.gunzip[IO]())
+      .through(Compression[IO].gunzip())
       .flatMap { result =>
         Stream.eval(IO {
           assertEquals(Some("payload.txt"), result.fileName)
@@ -265,7 +262,11 @@ class Fs2_core_3Test {
       _ <- thirdGate.complete(())
       _ <- secondGate.complete(())
       _ <- firstGate.complete(())
-      values <- fiber.join
+      values <- fiber.join.flatMap {
+        case Outcome.Succeeded(values) => values
+        case Outcome.Errored(error) => IO.raiseError(error)
+        case Outcome.Canceled() => IO.raiseError(new IllegalStateException("mapAsync stream was canceled"))
+      }
     } yield values).unsafeRunSync()
 
     assertEquals(Vector(1, 2, 3), ordered)
@@ -274,34 +275,30 @@ class Fs2_core_3Test {
   @Test
   @Timeout(value = 60, unit = TimeUnit.SECONDS)
   def queuesExposeBoundedOffersChunksAndTermination(): Unit = {
-    val queueResult: (Boolean, Boolean, Boolean, List[Int], Option[Int], Int, Int) = (for {
+    val queueResult: (Boolean, Boolean, Boolean, List[Int], Option[Int], Int) = (for {
       queue <- Queue.bounded[IO, Int](2)
-      firstOffer <- queue.offer1(1)
-      secondOffer <- queue.offer1(2)
-      thirdOffer <- queue.offer1(3)
-      chunk <- queue.dequeueChunk1(5)
-      empty <- queue.tryDequeue1
-      _ <- queue.enqueue1(4)
-      dequeued <- queue.dequeue1
-      mapped = queue.imap[String](_.toString)(_.toInt)
-      _ <- mapped.enqueue1("5")
-      mappedDequeued <- queue.dequeue1
+      firstOffer <- queue.tryOffer(1)
+      secondOffer <- queue.tryOffer(2)
+      thirdOffer <- queue.tryOffer(3)
+      chunk <- Stream.fromQueueUnterminated(queue, 5).take(2).compile.toList
+      empty <- queue.tryTake
+      _ <- queue.offer(4)
+      dequeued <- queue.take
     } yield (
       firstOffer,
       secondOffer,
       thirdOffer,
-      chunk.toList,
+      chunk,
       empty,
-      dequeued,
-      mappedDequeued
+      dequeued
     )).unsafeRunSync()
 
     val terminated: Vector[String] = (for {
-      queue <- Queue.noneTerminated[IO, String]
-      _ <- queue.enqueue1(Some("alpha"))
-      _ <- queue.enqueue1(Some("beta"))
-      _ <- queue.enqueue1(None)
-      values <- queue.dequeue.compile.toVector
+      queue <- Queue.unbounded[IO, Option[String]]
+      _ <- queue.offer(Some("alpha"))
+      _ <- queue.offer(Some("beta"))
+      _ <- queue.offer(None)
+      values <- Stream.fromQueueNoneTerminated(queue, 5).compile.toVector
     } yield values).unsafeRunSync()
 
     assertTrue(queueResult._1)
@@ -310,7 +307,6 @@ class Fs2_core_3Test {
     assertEquals(List(1, 2), queueResult._4)
     assertEquals(None, queueResult._5)
     assertEquals(4, queueResult._6)
-    assertEquals(5, queueResult._7)
     assertEquals(Vector("alpha", "beta"), terminated)
   }
 
@@ -328,7 +324,7 @@ class Fs2_core_3Test {
   private def byteStream(bytes: Array[Byte], chunkSize: Int): Stream[IO, Byte] = {
     val chunks: Vector[Chunk[Byte]] = bytes
       .grouped(chunkSize)
-      .map(chunk => Chunk.bytes(chunk))
+      .map(chunk => Chunk.array(chunk))
       .toVector
     Stream.emits(chunks).covary[IO].flatMap(chunk => Stream.chunk(chunk).covary[IO])
   }

--- a/tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/user-code-filter.json
+++ b/tests/src/co.fs2/fs2-core_3/3.0-130-1713a29/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "fs2.**"
+    },
+    {
+      "includeClasses" : "co_fs2.fs2_core_3.**"
+    }
+  ]
+}

